src/peergos/shared/user/fs/LazyInputStreamCombiner.java

```diff
@@ -26,7 +26,7 @@ public LazyInputStreamCombiner(FileRetriever stream, UserContext context, Symmet
         this.dataKey = dataKey;
         this.current = chunk;
         this.index = 0;
-        this.next = stream.getNext();
+        this.next = stream.getNext(dataKey).orElse(null);
```

Comment:
``orElse(null)``. Sadness, in the glorious land of Optionals can we not do away with null values?